quill-core/src/main/scala/io/getquill/quotation/Quotation.scala

```diff
@@ -54,4 +74,32 @@ trait Quotation extends Parsing with Liftables with Unliftables {
       annotation <- method.annotations.headOption
       astTree <- annotation.tree.children.lastOption
     } yield (astTree)
+
+  private class FreeVarTraverser extends Traverser {
```

Comment:
Let's do that in another PR. It won't be that easy, the `tree.symbol` is really powerful since you don't have to dig in the tree to find its symbol. With our AST, we would have to traverse down to the `Ident` and backtrack in order to find out if that block is a Free Variable or not.